src/oadp_resources/oadp/datamover/volume_snapshot_restore.py

```diff
@@ -1,6 +1,45 @@
 from ocp_resources.resource import NamespacedResource
+
+from oadp_constants.oadp.datamover.volume_snapshot_restore import VolumeSnapshotRestorePhase
 from oadp_constants.resources import ApiGroups
 
 
-class VolumeSnapshotLocation(NamespacedResource):
+class VolumeSnapshotRestore(NamespacedResource):
     api_group = ApiGroups.DATAMOVER_OADP_API_GROUP.value
+
+    def is_vsr_done(self):
+        return self.instance.status and self.instance.status.phase != \
+               VolumeSnapshotRestorePhase.SnapMoverRestorePhaseInProgress.value
+
+    @classmethod
+    def get_vsrs_by_restore_name(cls, restore_name):
```

Comment:
get_vsrs or it is only vsr ?

Reply:
it vsrs, that's why made is a classmethod

Reply:
do u get one vsr or a list of vsrs ?

Reply:
a list:
```python
        return list(cls.get(label_selector=f"velero.io/restore-name={restore_name}"))       
```

Reply:
I am a bit confused VSR should be a resource why we deal with lists ... ?

But if we really need it i would call it get_vsrs_by_label

Reply:
i thought we need some function like wait_delete wait_done , here it feels once we deal with one vsr like is_done and once we get all vsr by label which is list ...

Reply:
we have wait_for_status, wait_for_condtion and we can use the TImeSpmpler to wait for anything.
Please do not add any duplicated logic for that

Reply:
Exactly, yes 
Not need to implement the wait logic